### PR TITLE
Add a validated schedule expression to the Source schedule model

### PR DESCRIPTION
**Pull request by axmb**
Small meta-note: we'd debated not including things like schedule in our source schema, given that it'd be potentially duplicative of (and ergo inconsistent with) the data in AWS. For now I think this is the right thing to do, as it simplifies the work that the curator service does. Later, we should harden this synchronization.

### CODE REVIEW

File: verification/curator-service/api/src/model/schedule.ts [hunk 1]
```diff
@@ -6,8 +6,16 @@ export const scheduleSchema = new mongoose.Schema({
         required: 'Enter a CloudWatch schedule rule AWS Lambda ARN',
         match: [/^arn\:aws\:events\:.+\:.+\:rule\/.+/],
     },
+    awsScheduleExpression: {
+        type: String,
+        required: 'Enter a CloudWatch rule schedule expression',
+        match: [
+            /^((rate\(((1 (minute|hour|day))|([1-9]+ (minutes|hours|days)))\))|(cron\((.+\s){5}.+\)))/,
```
Comment (khmoran) [inline verification/curator-service/api/src/model/schedule.ts:17]: what a delicious regex. the comment is v. helpful :D

the match property is cool!

File: verification/curator-service/api/test/model/data/automation.full.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
         "awsLambdaArn": "arn:aws:lambda:region:id:function:name"
     },
     "schedule": {
-        "awsRuleArn": "arn:aws:events:region:id:rule/name"
+        "awsRuleArn": "arn:aws:events:region:id:rule/name",
+        "awsScheduleExpression": "cron(0 10 * * ? *)"
```
Comment (khmoran) [inline verification/curator-service/api/test/model/data/automation.full.json:7]: neither here nor there for this particular PR but wondering if we can actually use $ref so we don't have to re-specify sub-documents in these examples... not sure if they'll be dereferenced on import though... maybe something I'll try

i just wanna stay nice and DRY, you know?

File: verification/curator-service/api/test/model/schedule.test.ts [hunk 3]
```diff
@@ -25,6 +25,24 @@ describe('validate', () => {
...
+    it('a schedule without an AWS schedule expression is invalid', async () => {
+        const missingExpression = { ...fullModel };
+        delete missingExpression.awsScheduleExpression;
+
+        return new Schedule(missingExpression).validate((e) => {
+            expect(e.name).toBe(Error.ValidationError.name);
+        });
+    });
+
+    it('a schedule with a misformated AWS schedule expression is invalid', async () => {
```
Comment (khmoran) [inline verification/curator-service/api/test/model/schedule.test.ts:37]: super nit: s/misformated/misformatted

File: verification/curator-service/api/test/model/schedule.test.ts [hunk 4]
```diff
@@ -25,6 +25,24 @@ describe('validate', () => {
...
+        const badSchedule = { ...fullModel };
```
Comment (khmoran) [inline verification/curator-service/api/test/model/schedule.test.ts:38]: up to you, can inline as

`return new Schedule({ ...fullModel, awsScheduleExpression: 'rate(1 hour') }).validate ...`

but totally get it if you find that less readable

File: verification/curator-service/api/src/model/schedule.ts (hunk 1 above)
Comment (axmb) [reply verification/curator-service/api/src/model/schedule.ts:17]: Yeah -- I considered getting more precise with the `cron([...])` spec, but decided this was the right balance for now. Not that I don't love a good write-only regex.

File: verification/curator-service/api/test/model/data/automation.full.json (hunk 2 above)
Comment (axmb) [reply verification/curator-service/api/test/model/data/automation.full.json:7]: That'd be really neat -- let me know if you want a hand with that.

File: verification/curator-service/api/test/model/schedule.test.ts (hunk 3 above)
Comment (axmb) [reply verification/curator-service/api/test/model/schedule.test.ts:37]: I fear I'll never not misspell this.

File: verification/curator-service/api/test/model/schedule.test.ts (hunk 4 above)
Comment (axmb) [reply verification/curator-service/api/test/model/schedule.test.ts:38]: Ah, that's pretty nice! I'll shelf this for after the schema changes settle. This existing format is probably used in most tests.

**Comment by axmb** [comment]
[Add source automation schedule data to the ingestion source schema/API/UI](https://trello.com/c/xzbc0sdK/176-add-source-automation-schedule-data-to-the-ingestion-source-schema-api-ui)

**Pull request merged**